app/src/pages/playground/PlaygroundDatasetExamplesTable.tsx

```diff
@@ -116,9 +129,48 @@ const updateExampleResponsesMap = ({
   currentMap,
 }: {
   instanceId: number;
-  response: ChatCompletionResult | TextChunk | ToolCallChunk;
+  response:
+    | ChatCompletionResult
+    | TextChunk
+    | ToolCallChunk
+    | ChatCompletionOverDatasetMutationPayload;
   currentMap: InstanceToExampleResponsesMap;
 }): InstanceToExampleResponsesMap => {
+  if (response.__typename === "ChatCompletionOverDatasetMutationPayload") {
+    const instanceResponses: Record<string, ExampleRunData> = {};
```

Comment:
small thing: you basically want this to be the type on line 111 right, can use the same type / aliases as used up there here?
```
Record<ExampleId, ExampleRunData | undefined>
```

also you could consider splitting this function updateExampleResponsesFromSubscription and updateExampleResponsesFromMutation or somehing, those can be wrapped if you still want one but just cause this is getting so long and they don't share any logic (two separate switches)